src/api/auth.ts

```diff
@@ -42,34 +42,45 @@ export const logout = async () => {
   }
 };
 
-export const register = async (email: string, password: string) => {
+export const register = async (user: { email: string; password: string; firstName: string; lastName: string; address: string; country: string; city: string; }) => {
+  try {
     const response = await fetch(`${apiUrl}/register`, {
-        method: 'POST',
-        headers: {
-            'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ email, password }),
```

Comment:
Making register function to take only one argument instead of two will improve code readability.